.github/workflows/tfplugindocs.yaml

```diff
@@ -0,0 +1,48 @@
+---
+name: Terraform plugin docs
+concurrency: tfplugindocs
+on:
+    push:
+        branches:
+            - master
+
+permissions:
+    contents: write
+
+jobs:
+    tfplugindocs:
+        name: Generate Terraform plugin docs
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check out code
+              uses: actions/checkout@v4
+
+            - name: Set up Go
+              uses: actions/setup-go@v4
+              with:
+                  go-version-file: go.mod
+
+            - name: Generate docs
+              run: |
+                  make install-tools tfdocs_generate
+
+            - name: Fix trailing whitespace in generated docs
+              uses: pre-commit/action@v3.0.0
+              # ignoring errors, as pre-commit will fail if files were modified
+              continue-on-error: true
+              with:
+                  extra_args: trailing-whitespace --all-files
+
+            - name: Fix end of file in generated docs
+              uses: pre-commit/action@v3.0.0
+              # ignoring errors, as pre-commit will fail if files were modified
+              continue-on-error: true
+              with:
+                  extra_args: end-of-file-fixer --all-files
```

Comment:
What about this:

**`pre-commit-config.yaml`**

```diff
repos:
    - repo: https://github.com/pre-commit/pre-commit-hooks
      rev: v4.4.0
      hooks:
          - id: trailing-whitespace
+           stages: [commit, push, manual]
          - id: end-of-file-fixer
+           stages: [commit, push, manual]
```

And then `extra_args` would be `--hook-stage=manual --all-files`

Reply:
Good suggestion. I didn't configure pre-commit before, so didn't know about stages.